central/complianceoperator/v2/checkresults/datastore/stats_views.go

```diff
@@ -58,15 +60,17 @@ type ResourceResultCountByProfile struct {
 
 // ResourceResultsByProfile represents shape of the stats query for compliance operator results
 type ResourceResultsByProfile struct {
-	PassCount          int    `db:"pass_count"`
-	FailCount          int    `db:"fail_count"`
-	ErrorCount         int    `db:"error_count"`
-	InfoCount          int    `db:"info_count"`
-	ManualCount        int    `db:"manual_count"`
-	NotApplicableCount int    `db:"not_applicable_count"`
-	InconsistentCount  int    `db:"inconsistent_count"`
-	ProfileName        string `db:"compliance_profile_name"`
-	CheckName          string `db:"compliance_check_name"`
-	RuleName           string `db:"compliance_rule_name"`
-	CheckRationale     string `db:"compliance_check_rationale"`
+	PassCount          int                        `db:"pass_count"`
+	FailCount          int                        `db:"fail_count"`
+	ErrorCount         int                        `db:"error_count"`
+	InfoCount          int                        `db:"info_count"`
+	ManualCount        int                        `db:"manual_count"`
+	NotApplicableCount int                        `db:"not_applicable_count"`
+	InconsistentCount  int                        `db:"inconsistent_count"`
+	ProfileName        string                     `db:"compliance_profile_name"`
+	CheckName          string                     `db:"compliance_check_name"`
+	RuleName           string                     `db:"compliance_rule_name"`
+	RuleId             string                     `db:"compliance_rule_id"`
+	CheckRationale     string                     `db:"compliance_check_rationale"`
+	Controls           []*datastore.ControlResult `db:"compliance_control"`
```

Comment:
Is it okay to add a field here which is added **not** by the framework?
I would think this is not a good pattern to enhance it afters, but not sure on this though.

Reply:
I would not add this here.  I would get the control data separately and pass it to the convertor to put in the API response.